docs/near_network.md

```diff
@@ -0,0 +1,65 @@
+# 1. Overview
+Near Protocol uses its own implementation of a custom peer-to-peer network
+Peers who join network are represented by nodes and connected between them by edges.
+
+The purpose of this document is to describe inner workings of `near-network` package;
+and to be used as reference by future engineers to understand network code without any prior knowledge.
```

Comment:
:+1: for stating the purpose&target audience of the doc! It goes a long way towards keeping the thing maintainable over the long run.